docs/rules/no-undef-components-in-script-setup.md

```diff
@@ -0,0 +1,76 @@
+---
+pageClass: rule-details
+sidebarDepth: 0
+title: vue/no-undef-components-in-script-setup
+description: disallow undefined components in `<template>` with `<script setup>`
+---
+# vue/no-undef-components-in-script-setup
+
+> disallow undefined components in `<template>` with `<script setup>`
+
+- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> ***This rule has not been released yet.*** </badge>
+
+## :book: Rule Details
+
+This rule warns that the component used in the `<template>` is not defined in `<script setup>`.  
+
+Undefined components in `<script setup>` will be resolved from the components registered in the global. However, if you are not using global components, you can use this rule to prevent run-time errors.
+
+<eslint-code-block :rules="{'vue/no-undef-components-in-script-setup': ['error']}">
+
+```vue
+<script setup>
+import Foo from './Foo.vue'
+</script>
+
+<template>
+  <!-- ✓ GOOD -->
+  <Foo />
+
+  <!-- ✗ BAD -->
+  <Bar />
+</template>
+```
+
+</eslint-code-block>
+
+## :wrench: Options
+
+```json
+{
+  "vue/no-undef-components-in-script-setup": ["error", {
+    "ignorePatterns": []
+  }]
+}
+```
+
+- `ignorePatterns` Suppresses all errors if component name matches one or more patterns.
+
+### `ignorePatterns: ['custom(\\-\\w+)+']`
+
+<eslint-code-block :rules="{'vue/no-undef-components-in-script-setup': ['error', { 'ignorePatterns': ['custom(\\-\\w+)+'] }]}">
+
+```vue
+<script setup>
+</script>
+
+<template>
+  <!-- ✓ GOOD -->
+  <CustomComponent />
+
+  <!-- ✗ BAD -->
+  <Bar />
+</template>
+```
+
+</eslint-code-block>
+
+## :couple: Related Rules
+
+- [vue/no-unregistered-components](./no-unregistered-components.md)
```

Comment:
Why are these rules separate? They report the same issues, but only use different places to determine which components are registered/defined. If it's possible to merge them, I'd vote for that.

Reply:
I think the word "unregistered" is not correct because the script setup doesn't register the component. So I thought it should be a different rule.
But as you say, I think the rules are reporting the same issue. It may make more sense to mark `vue/no-unregistered-components` as deprecated and merge the new rule as `vue/no-undef-components`.
Do you think it's better to deprecate the old rule and create a new one?

Reply:
Yeah, that sound fine.

Reply:
I will change this PR later :+1: